mjpc/tasks/allegro/allegro.cc

```diff
@@ -0,0 +1,101 @@
+#include "mjpc/tasks/allegro/allegro.h"
+#include <iostream>
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include <mujoco/mujoco.h>
+#include "mjpc/task.h"
+#include "mjpc/utilities.h"
+
+namespace mjpc {
+std::string Allegro::XmlPath() const {
+  return GetModelPath("allegro/task.xml");
+}
+std::string Allegro::Name() const { return "Allegro"; }
+
+// ------- Residuals for cube manipulation task ------
+//     Cube position: (3)
+//     Cube orientation: (3)
+//     Cube linear velocity: (3)
+//     Control:  u
+// ------------------------------------------
+void Allegro::ResidualFn::Residual(const mjModel* model, const mjData* data,
+                        double* residual) const {
+  int counter = 0;
+
+  // ---------- Cube position ----------
+  // TODO(vincekurtz): specify goal position in a better way
+  double* cube_position = SensorByName(model, data, "cube_position");
+  std::vector<double> goal_cube_position = {0.3, 0.0, 0.025};
+
+  mju_sub3(residual + counter, cube_position, goal_cube_position.data());
+  counter += 3;
+
+  // ---------- Cube orientation ----------
+  double* cube_orientation = SensorByName(model, data, "cube_orientation");
+  double* goal_cube_orientation = SensorByName(model, data, "cube_goal_orientation");
+  mju_normalize4(goal_cube_orientation);
+
+  mju_subQuat(residual + counter, goal_cube_orientation, cube_orientation);
+  counter += 3;
+
+  // ---------- Cube linear velocity ----------
+  double* cube_linear_velocity =
+      SensorByName(model, data, "cube_linear_velocity");
+
+  mju_copy(residual + counter, cube_linear_velocity, 3);
+  counter += 3;
+
+  // ---------- Control ----------
+  mju_copy(residual + counter, data->actuator_force, model->nu);
+  counter += model->nu;
+
+  // ---------- Residual () ----------
```

Comment:
I left these off... should be labeled Nominal pose and Joint velocity. Update the Residual description above.